packages/core/src/components/editable-text/editableText.tsx

```diff
@@ -397,6 +424,33 @@ export class EditableText extends AbstractPureComponent2<IEditableTextProps, IEd
             }
         }
     }
+
+    private maybeRenderRightElement() {
+        const { rightElement } = this.props;
+        if (rightElement == null) {
+            return undefined;
+        }
+        return (
+            <span
+                className={Classes.EDITABLE_TEXT_ACTION}
+                onMouseEnter={this.handleMouseEnter}
+                onMouseLeave={this.handleMouseLeave}
+                onClick={this.handleFocus}
+                ref={this.refHandlers.rightElement}
+            >
+                {rightElement}
+            </span>
+        );
+    }
+
+    private handleMouseEnter = () => {
+        this.setState({ isRightElementVisible: true });
```

Comment:
what is the UX logic here? I was sort of expecting the right element to become visible once mousing into the whole editable text, not just the right element. also, why are we focusing the input at this time?